enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
-As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
-AWS resource tags.
+### API Extensions
+
+This proposal edits `Infrastructure.config.openshift.io/v1` type
+
+### Operational Aspects of API Extensions
+
+NA
 
 ### Risks and Mitigations
 
+#### Failure Modes
+
+NA
+
+#### Support Procedures
+
+NA
+
 ## Design Details
 
+User can add tags during installation for creation of resources with tags. Installer creates infrastructure manifests with `resourceTags` to `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io`.
+Operators that operate on aws resources must consider the tags from `.spec.platformSpec.aws`, `.status.platformStatus.aws` and existing AWS resource tags.
+
+The values from `status.platformStatus.aws` will used to only support older version.
+
+`.status.platformStatus.aws` will be deprecated in the future versions.
+
 ### Test Plan
 
-TBD
+Update the `resourceTags` testcases to check successful update from Infrastructure resource.
 
 ### Graduation Criteria
 
-TBD
-
 #### Dev Preview -> Tech Preview
 
-TBD
+NA
 
 #### Tech Preview -> GA
 
-TBD
+- Upgrade/downgrade testing
+- Sufficient time for feedback
+- Available by default
+- Stress testing for scaling and tag update scenarios
 
 #### Removing a deprecated feature
 
-This enhancement does not remove or deprecate any features.
+`experimentalPropagateUserTags` field will be set for deprecation.
+`.status.platformStatus.aws` will be set for deprecation.
 
 ### Upgrade / Downgrade Strategy
 
 On upgrade:
 
-The new status field won't be populated since it is only populated by the installer and that can't have happened if the cluster was installed from a prior version. Components that consume the new field should take no action since they will see no additional tags.
+On upgrade from version supporting `experimentalPropagateUserTags`, openshift components that consume the new field should merge `.status.platformStatus.aws` and `.spec.platformSpec.aws`.
+`.spec.platformSpec.aws` can be updated or changed which will override the values from status.
+
+For installer configuration compatibility, configuration with `experimentalPropagateUserTags` should be supported.
 
 On downgrade:
 
-The status field may remain populated, components may or may not continue to tag newly created resources w/ the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
+The status field may remain populated, components may or may not continue to tag newly created resources with the additional tags depending on whether or not a given component still has logic to respect the status tags, after the downgrade.
 
-### Version Skew Strategy
+`experimentalPropagateUserTags` field should be generated in installer configuration to support lower version installers.
```

Comment:
What does this mean?

Reply:
This means, the install-config generated with the proposed version can still be used with experimental version which adds tags to status field

Reply:
We generally do not support taking an install config from version n+1 and using it with a installer creating a cluster of version n.

Reply:
Ok, updated EP to remove the lower version compatibility.